### PR TITLE
refactor(sources): extract source content renderer

### PR DESCRIPTION
**Pull request by teng-lin**
## Summary
- move source fulltext and guide RPC/parsing behavior into SourceContentRenderer
- keep SourcesAPI get_fulltext, get_guide, and _extract_all_text as compatibility delegators
- add direct renderer coverage for text/markdown params, missing markdownify, HTML/plaintext parsing, source-not-found, logging, recursion guard, URL/type parsing, and guide shape variants

## Verification
- rtk uv run pytest tests/unit/test_source_content_renderer.py tests/unit/test_sources_integration.py tests/unit/cli/test_source.py::TestSourceFulltext tests/unit/cli/test_source.py::TestSourceGuide tests/unit/test_json_stdout_purity.py tests/integration/cli_vcr/test_sources.py::TestSourceContentCommands tests/integration/test_vcr_comprehensive.py::TestSourcesAPI::test_get_fulltext tests/integration/test_vcr_comprehensive.py::TestSourcesAPI::test_get_guide
- rtk uv run pytest tests/unit/test_init_order.py tests/unit/test_capabilities.py -q
- rtk uv run pytest tests/unit/test_api_coverage.py::TestGetSourceGuide -q
- rtk uv run ruff check src/notebooklm/_sources.py src/notebooklm/_source_content.py tests/unit/test_source_content_renderer.py tests/unit/test_sources_integration.py tests/unit/cli/test_source.py tests/unit/test_json_stdout_purity.py tests/integration/cli_vcr/test_sources.py tests/integration/test_vcr_comprehensive.py
- rtk uv run ruff format --check src/notebooklm/_sources.py src/notebooklm/_source_content.py tests/unit/test_source_content_renderer.py tests/unit/test_sources_integration.py tests/unit/cli/test_source.py tests/unit/test_json_stdout_purity.py tests/integration/cli_vcr/test_sources.py tests/integration/test_vcr_comprehensive.py
- rtk uv run mypy src/notebooklm
- rtk git diff --check

## Summary by CodeRabbit

* **New Features**
  * Source content rendering: fetches guide summaries/keywords and full-text in text or markdown (converts HTML when available).
* **Bug Fixes / Reliability**
  * Robust parsing of nested content with depth guards; warnings for empty/missing renditions.
  * Better validation and clear errors for invalid formats and missing sources.
* **Tests**
  * Added unit tests covering rendering modes, markdown dependency failure, parsing edge cases, and nesting/max-depth behavior.

[Review Change Stack](https://app.coderabbit.ai/change-stack/teng-lin/notebooklm-py/pull/684?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Comment by teng-lin** [comment]
@claude review

**Comment by teng-lin** [comment]
Addressed CodeRabbit's redundant-guard nitpick in 6d8d770: removed the already-proven result/list guard and kept the existing shape checks. Re-verified the focused T9f suite, ruff, format, and mypy.

File: src/notebooklm/_source_content.py [hunk 1]
```diff
@@ -1,10 +1,161 @@
...
+                    source_id,
+                    source_type,
+                )
+        else:
+            if len(result) > 3 and isinstance(result[3], list) and len(result[3]) > 0:
+                content_blocks = result[3][0]
+                if isinstance(content_blocks, list):
+                    texts = self.extract_all_text(content_blocks)
+                    content = "\n".join(texts)
+
```
[the remark this answers is not in the document]
Comment (teng-lin) [reply src/notebooklm/_source_content.py]: Addressed in ec1ceb8: removed the now-redundant length check while keeping the result[0] list/shape checks. Re-verified focused T9f suite, ruff, format, and mypy.

**Comment by teng-lin** [comment]
@claude review
